File: mercury mobile/lib/data/categories.dart
import 'package:flutter/material.dart';
import 'package:iconsax_plus/iconsax_plus.dart';

import '../models/category.dart';

/// The focused departments, aligned with the web taxonomy. `slug` matches the
/// product `categoryId` in Firestore (laptops, desktops, printers-office,
/// networking-security, ups-power, software, other). Subcategories here are
/// sensible fallbacks; the live list is merged from Firestore `children`.
const List<Category> kShopCategories = [
  Category(
    name: 'Laptops',
    slug: 'laptops',
    color: Color(0xFF1F3E97),
    image: 'assets/images/computers-removebg-preview.png',
    photo: 'assets/images/cat-laptops.jpeg',
    imageScale: 1.8,
    imageScaleX: 2.15,
    imageRotationDeg: -90,
    subcategories: [
      Subcategory(
        'All',
        IconsaxPlusBold.category_2,
        image: 'assets/images/computers-removebg-preview.png',
      ),
      Subcategory(
        'HP Laptops',
        IconsaxPlusBold.monitor,
        image: 'assets/images/HP 250 G9 Laptop.jpeg',
      ),
      Subcategory('Lenovo Laptops', IconsaxPlusBold.monitor),
      Subcategory('Dell Laptops', IconsaxPlusBold.monitor),
      Subcategory('Gaming Laptops', IconsaxPlusBold.monitor),
      Subcategory('Business Laptops', IconsaxPlusBold.monitor),
    ],
  ),
  Category(
    name: 'Desktops',
    slug: 'desktops',
    color: Color(0xFF0F766E),
    image: 'assets/images/computers-removebg-preview.png',
    photo: 'assets/images/cat-desktops.jpeg',
    subcategories: [
      Subcategory(
        'Desktops',
        IconsaxPlusBold.devices,
        image: 'assets/images/Dell OptiPlex 7020 MT (desktop + monitor).jpeg',
      ),
      Subcategory(
        'All',
        IconsaxPlusBold.category_2,
        image: 'assets/images/computers-removebg-preview.png',
      ),
      Subcategory('Dell Desktops', IconsaxPlusBold.devices),
      Subcategory('HP Desktops', IconsaxPlusBold.devices),
      Subcategory(
        'Monitors',
        IconsaxPlusBold.monitor_mobbile,
        image: 'assets/images/Dell E2020H.jpeg',
      ),
    ],
  ),
  Category(
    name: 'Printers & Office',
    slug: 'printers-office',
    color: Color(0xFFD9620E),
    image: 'assets/images/printers___power-removebg-preview.png',
    photo: 'assets/images/cat-office.jpeg',
    imageScale: 1.3,
    subcategories: [
      Subcategory(
        'Printers',
        IconsaxPlusBold.printer,
        image: 'assets/images/HP DeskJet 2320.jpeg',
      ),
      Subcategory(
        'All',
        IconsaxPlusBold.category_2,
        image: 'assets/images/printers___power-removebg-preview.png',
      ),
      Subcategory('Ink & Toner', IconsaxPlusBold.colorfilter),
      Subcategory('Scanners', IconsaxPlusBold.scanner),
    ],
  ),
  Category(
    name: 'Networking & Security',
    slug: 'networking-security',
    color: Color(0xFF0E7490),
    image: 'assets/images/networking___security-removebg-preview.png',
    photo: 'assets/images/cat-networking.jpeg',
    imageScale: 1.45,
    subcategories: [
      Subcategory(
        'All',
        IconsaxPlusBold.category_2,
        image: 'assets/images/networking___security-removebg-preview.png',
      ),
      Subcategory('Routers', IconsaxPlusBold.routing),
      Subcategory('Switches', IconsaxPlusBold.routing),
      Subcategory('Wi-Fi', IconsaxPlusBold.wifi),
      Subcategory('CCTV', IconsaxPlusBold.camera),
    ],
  ),
  Category(
    name: 'UPS & Power',
    slug: 'ups-power',
    color: Color(0xFF1E293B),
    image: 'assets/images/components___power-removebg-preview.png',
    photo: 'assets/images/cat-components.jpeg',
    subcategories: [
      Subcategory(
        'All',
        IconsaxPlusBold.category_2,
        image: 'assets/images/components___power-removebg-preview.png',
      ),
      Subcategory('UPS', IconsaxPlusBold.battery_full),
      Subcategory('Power', IconsaxPlusBold.flash),
    ],
  ),
  Category(
    name: 'Software',
    slug: 'software',
    color: Color(0xFF5B21B6),
    image: 'assets/images/cat-software.png',
    photo: 'assets/images/cat-software.jpeg',
    imageScale: 1.3,
    subcategories: [
      Subcategory(
        'All',
        IconsaxPlusBold.category_2,
        image: 'assets/images/accessories-removebg-preview.png',
      ),
      Subcategory('Computer Software', IconsaxPlusBold.document_text),
      Subcategory('Microsoft 365 Family', IconsaxPlusBold.document_text),
    ],
  ),
  Category(
    name: 'Other Products',
    slug: 'other',
    color: Color(0xFF9F1239),
    image: 'assets/images/accessories-removebg-preview.png',
    photo: 'assets/images/cat-accessories.jpeg',
    imageScale: 1.45,
    subcategories: [
      Subcategory(
        'All',
        IconsaxPlusBold.category_2,
        image: 'assets/images/accessories-removebg-preview.png',
      ),
      Subcategory('Phones', IconsaxPlusBold.mobile),
      Subcategory('Tablets', IconsaxPlusBold.mobile),
      Subcategory('Audio', IconsaxPlusBold.headphone),
      Subcategory('Accessories', IconsaxPlusBold.mouse),
    ],
  ),
];

File: web/src/lib/categories.ts
export type SubCategory = {
  name: string;
  slug: string;
};

export type Category = {
  name: string;
  slug: string;
  image: string;
  children: SubCategory[];
};

const sub = (name: string, slug: string): SubCategory => ({ name, slug });

/// Full category taxonomy mapped to actual product data from the old site.
/// `children` are the subcategories that match the `category` field in Firestore.
export const CATEGORIES: Category[] = [
  {
    name: "Computers",
    slug: "computers",
    image: "/cat-computers.jpeg",
    children: [
      sub("Lenovo Laptops", "lenovo-laptops"),
      sub("HP Laptops", "hp-laptops"),
      sub("Dell Laptops", "dell-laptops"),
      sub("Business Laptops", "business-laptops"),
      sub("Gaming Laptops", "gaming-laptops"),
      sub("2-in-1 Laptops", "2-in-1-laptops"),
      sub("Desktops", "desktops"),
      sub("Dell Desktops", "dell-desktops"),
      sub("HP Desktops", "hp-desktops"),
      sub("Monitors", "monitors"),
      sub("Samsung Tablets", "samsung-tablets"),
      sub("Servers", "servers"),
    ],
  },
  {
    name: "Printers & Office",
    slug: "printers-office",
    image: "/cat-office.jpeg",
    children: [
      sub("All-in-One Printers", "multifunction-all-in-one-printers"),
      sub("Ink Tank Printers", "ink-tank-printers"),
      sub("Laser Printers", "a4-black-white-laser-printers"),
      sub("Color Laser Printers", "color-laser-multifunction-printers"),
      sub("Photo Printers", "photo-printers"),
      sub("HP Toner Cartridges", "hp-toner-cartridges"),
      sub("Compatible Toner", "compatible-toner-cartridge"),
      sub("HP Ink Cartridges", "hp-ink-cartridges"),
      sub("Scanners", "hp-scanners"),
      sub("Paper Shredders", "rexel-paper-shredders"),
      sub("Money Counting Machines", "money-counting-machines"),
    ],
  },
  {
    name: "Components & Power",
    slug: "components-power",
    image: "/cat-components.jpeg",
    children: [
      sub("APC UPS", "apc-ups"),
      sub("APC Smart UPS", "apc-smart-ups"),
      sub("APC Easy UPS", "apc-easy-ups"),
      sub("Giganet UPS", "giganet-ups"),
      sub("Laptop RAM", "laptop-ram"),
      sub("Portable SSD", "portable-ssd"),
      sub("Storage", "storage"),
    ],
  },
  {
    name: "Networking & Security",
    slug: "networking-security",
    image: "/cat-networking.jpeg",
    children: [
      sub("Networking", "networking"),
      sub("Routers", "routers"),
      sub("4G Routers", "4g-routers"),
      sub("Switches", "switches"),
      sub("Hikvision Cameras", "hikvision-cameras"),
      sub("Bullet Cameras", "bullet-cameras"),
      sub("Security Systems", "security-systems"),
      sub("Wi-Fi Adapters", "wi-fi-adapters"),
    ],
  },
  {
    name: "Phones, TV & Audio",
    slug: "phones-tv-audio",
    image: "/cat-phones.jpeg",
    children: [
      sub("Apple iPhone", "apple-iphone"),
      sub("Nokia Phones", "nokia-phones"),
      sub("Smart TV", "smart-tv"),
      sub("Headphones", "headphones"),
      sub("Headsets", "headsets"),
      sub("Jabra Headsets", "jabra-headsets"),
      sub("Speakers", "speakers"),
      sub("Conference Cameras", "conference-camera"),
      sub("Projectors", "epson-projector"),
    ],
  },
  {
    name: "Accessories",
    slug: "accessories",
    image: "/cat-accessories.jpeg",
    children: [
      sub("Accessories", "accessories"),
      sub("Webcams", "webcams"),
      sub("Wireless Mouse", "wireless-mouse"),
      sub("Laptop Chargers", "laptop-chargers"),
      sub("Gaming Consoles", "gaming-consoles"),
      sub("Computer Software", "computer-software"),
      sub("Microsoft Licensing", "microsoft-365-family"),
    ],
  },
];
